lib/src/main/java/graphql/nadel/validation/NadelHydrationArgumentValidation.kt

```diff
@@ -88,8 +90,9 @@ internal class NadelHydrationArgumentValidation private constructor() {
                 hydration: UnderlyingServiceHydration
         ): NadelSchemaValidationError? {
 
-            //need to check null compatibility for inner types (as this is recursive)
-            if (actorFieldArgType.isNonNull && !hydrationSourceFieldType.isNonNull) {
+            //need to check null compatibility
+            val sourceType = remoteArg.remoteArgumentSource.sourceType
+            if (sourceType != ObjectField && actorFieldArgType.isNonNull && !hydrationSourceFieldType.isNonNull) {
```

Comment:
added `sourceType != ObjectField ` so that it ignores null compatibility for `$source` arguments as that is handled differently by nadel